lib/Basic/Platform.cpp

```diff
@@ -118,6 +118,12 @@ StringRef swift::getPlatformNameForTriple(const llvm::Triple &triple) {
   case llvm::Triple::FreeBSD:
     return "freebsd";
   case llvm::Triple::Win32:
+    if (triple.isWindowsCygwinEnvironment())
+      return "cygwin";
+
+    if (triple.isWindowsGNUEnvironment())
+      return "mingw";
+
```

Comment:
I think it would be nicer to use a switch here instead.  That way if we run into similar issues in the future, its easier to extend.